extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -742,6 +749,31 @@ export class AmpStoryPage extends AMP.BaseElement {
         eventInit);
   }
 
+  /**
+  * Checks if the page has any audio
+  * @private
+  */
+  checkPageHasAudio_() {
+    const hasBGPageAudio = this.element.hasAttribute('background-audio');
+    if (hasBGPageAudio) {
+      this.storeService_.dispatch(Action.TOGGLE_PAGE_HAS_AUDIO, true);
+      return;
+    }
+
+    const hasAmpAudio = this.element.querySelector('amp-audio');
+    if (hasAmpAudio) {
+      this.storeService_.dispatch(Action.TOGGLE_PAGE_HAS_AUDIO, true);
+      return;
+    }
+
+    const videos = this.element.querySelectorAll('amp-video');
+    const hasVideoWithAudio = Array.prototype.some.call(videos, video => {
+      return !video.hasAttribute('noaudio');
+    });
+
+    this.storeService_.dispatch(Action.TOGGLE_PAGE_HAS_AUDIO,
+        hasVideoWithAudio);
+  }
```

Comment:
Could we factorize this a bit?

What do you think of:

````
checkPageHasAudio_() {
  const pageHasAudio =
      this.element.hasAttribute('background-audio') ||
      this.element.querySelector('amp-audio') ||
      this.hasVideoWithAudio_();

  this.storeService_.dispatch(Action.TOGGLE_PAGE_HAS_AUDIO, pageHasAudio);
}

hasVideoWithAudio_() {
    const ampVideoEls = this.element.querySelectorAll('amp-video');
    return Array.prototype.some.call(ampVideoEls, video => !video.hasAttribute('noaudio'));
}
````

Chaining the `||` will exit as soon as one expression is truthy, so we would keep queries to the DOM minimal just like in your implementation.

Reply:
Ooh that works great! And it looks a lot better haha. thanks!